types/hapi-auth-cookie/index.d.ts

```diff
@@ -0,0 +1,124 @@
+     */
+    interface Options {
+        /**
+         * The cookie name. Defaults to 'sid'
+         */
+        cookie?: string;
+
+        /**
+         * Used for Iron cookie encoding.
+         * Should be at least 32 characters long.
+         */
+        password: string;
+
+        /**
+         * Sets the cookie expires time in milliseconds.
+         * Defaults to single browser session (ends when browser closes).
+         * Required when keepAlive is true.
+         */
+        ttl?: number;
+
+        /**
+         * Sets the cookie Domain value.
+         * Defaults to none.
+         */
+        domain?: string;
+
+        /**
+         * Sets the cookie path value.
+         * Defaults to /.
+         */
+        path?: string;
+
+        /**
+         * Any authentication cookie that fails validation will be marked as expired in the response and cleared.
+         * Defaults to false.
+         */
+        clearInvalid?: boolean;
+
+        /**
+         * Automatically sets the session cookie after validation to extend the current session for a new ttl duration.
+         * Defaults to false.
+         */
+        keepAlive?: boolean;
+
+        /**
+         * If false omitted.
+         * Other options Strict or Lax.
+         * Defaults to Strict.
+         */
+        isSameSite?: boolean | string;
+
+        /**
+         * If false, the cookie is allowed to be transmitted over insecure connections which exposes it to attacks.
+         * Defaults to true.
+         */
+        isSecure?: boolean;
+
+        /**
+         * If false, the cookie will not include the 'HttpOnly' flag.
+         * Defaults to true.
+         */
+        isHttpOnly?: boolean;
+
+        /**
+         * Login URI or function that returns a URI to redirect unauthenticated requests to.
+         * Note that it will only trigger when the authentication mode is 'required'.
+         * Defaults to no redirection.
+         */
+        redirectTo?: string | redirectToFunction;
+
+        /**
+         * Only works if `redirectTo` is true
+         * If set to true, a string, or an object, appends the current request path to the query component of the redirectTo URI
+         */
+        appendNext?: boolean | string;
+
+        /**
+         * an optional session validation function used to validate the content of the session cookie on each request.
+         * Used to verify that the internal session state is still valid (e.g. user account still exists).
+         */
+        validateFunc?: validateFunction;
+
+        /**
+         * A name to use with decorating the request object.
+         * Defaults to 'cookieAuth'.
+         * Using multiple decorator names for separate authentication strategies could allow a developer to call the methods for the wrong strategy.
+         * Potentially resulting in unintended authorized access.
+         */
+        requestDecoratorName?: string;
+    }
+}
+
+declare const hapiAuthCookie: Plugin<{}>;
```

Comment:
It doesn't take any arguments when registering as far as I know. Continue using this empty object? Or `null` | `false`? What is the recommended type to use in that case?

Reply:
Would use `void`.